src/components/send/form/formBtc.js

```diff
@@ -0,0 +1,188 @@
+import { fromRawLsk, toRawLsk } from '../../../utils/lsk';
+import FormBase from './formBase';
+import Selector from '../../toolbox/selector/selector';
+import Spinner from '../../spinner/spinner';
+import Tooltip from '../../toolbox/tooltip/tooltip';
+import styles from './form.css';
+
+function getInitialState() {
+  return {
+    isLoading: false,
+    fields: {
+      processingSpeed: {
+        value: 0,
+        loaded: false,
+        txFee: 0,
+        selectedIndex: 0,
+      },
+      amount: {
+        value: '',
+      },
+      fee: {
+        value: 0,
+      },
+    },
+    unspentTransactionOutputs: [],
+  };
+}
+
+export default class FormBtc extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = getInitialState();
+
+    this.getProcessingSpeedStatus = this.getProcessingSpeedStatus.bind(this);
+    this.selectProcessingSpeed = this.selectProcessingSpeed.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.dynamicFeesRetrieved();
+  }
+
+  componentDidUpdate() {
+    const { fields, unspentTransactionOutputs } = this.state;
+    const {
+      token, account, dynamicFees, networkConfig,
+    } = this.props;
+    // istanbul ignore next
+    if (!Object.keys(dynamicFees).length) this.props.dynamicFeesRetrieved();
+    if (account && account.info[token]
+        && !unspentTransactionOutputs.length) {
+      btcTransactionsAPI
+        .getUnspentTransactionOutputs(account.info[token].address, networkConfig)
+        .then(data => this.setState(() => ({ unspentTransactionOutputs: data })))
+        .catch(() => this.setState(() => ({ unspentTransactionOutputs: [] })));
+    }
+
+    // istanbul ignore if
+    if (!fields.processingSpeed.loaded && dynamicFees.Low) {
+      this.setState(() => ({
+        fields: {
+          ...fields,
+          processingSpeed: {
+            value: dynamicFees.Low,
+            loaded: true,
+            txFee: this.getCalculatedDynamicFee(dynamicFees.Low),
+          },
+        },
+      }));
+    }
+  }
+
+  // istanbul ignore next TODO add test for this
+  onInputChange({ target }, newState) {
+    const { fields } = this.state;
+    if (target.name === 'amount') {
+      const txFee = this.getCalculatedDynamicFee(fields.processingSpeed.value, target.value);
+      this.setState(() => ({
+        fields: {
+          ...fields,
+          processingSpeed: {
+            ...fields.processingSpeed,
+            txFee,
+          },
+          [target.name]: newState,
+        },
+      }));
+    }
+  }
+
+  /**
+   * Get status of processing soeed fetch based on state of component
+   * @returns {Node} - Text to display to the user or loader
+   */
+  // istanbul ignore next TODO add test for this
+  getProcessingSpeedStatus() {
+    const { token, t } = this.props;
+    const { fields, isLoading } = this.state;
+    const { amount } = fields;
```

Comment:
here you can do the same that you did in line 119

`const { fields: { amount } } = this.state;`

Reply:
I tried that, but it's not possible - I get `no-undef: 'fields' is not defined.` on line `112`

Reply:
ok no problem all the code is good is a minor thing